src/native/eventpipe/ep-session.c

```diff
@@ -296,12 +299,22 @@ ep_session_remove_dangling_session_states (EventPipeSession *session)
 }
 
 void
-ep_session_free (EventPipeSession *session)
+ep_session_inc_ref (EventPipeSession *session)
+{
+	session->ref_count++;
+}
+
+void
+ep_session_dec_ref (EventPipeSession *session)
 {
 	ep_return_void_if_nok (session != NULL);
 
 	EP_ASSERT (!ep_session_get_streaming_enabled (session));
 
+	session->ref_count--;
```

Comment:
I think we should make this thread safe, so ep_rt_atomic_dec_uint32_t in this case.

Reply:
```
if (ep_rt_atomic_dec_uint32_t(&session->ref_count)  != 0)
    return;
```